www/js/diary/timelineHelper.ts

```diff
@@ -211,3 +215,348 @@ const locations2GeojsonTrajectory = (trip, locationList, trajectoryColor?) => {
     };
   });
 };
+
+// DB entries retrieved from the server have '_id', 'metadata', and 'data' fields.
+// This function returns a shallow copy of the obj, which flattens the
+// 'data' field into the top level, while also including '_id' and 'metadata.key'
+const unpackServerData = (obj: ServerData<any>) => ({
+  ...obj.data,
+  _id: obj._id,
+  key: obj.metadata.key,
+  origin_key: obj.metadata.origin_key || obj.metadata.key,
+});
+
+export const readAllCompositeTrips = function (startTs: number, endTs: number) {
+  const readPromises = [getRawEntries(['analysis/composite_trip'], startTs, endTs, 'data.end_ts')];
+  return Promise.all(readPromises)
+    .then(([ctList]: [ServerResponse<TimelineEntry>]) => {
+      return ctList.phone_data.map((ct) => {
+        const unpackedCt = unpackServerData(ct);
+        return {
+          ...unpackedCt,
+          start_confirmed_place: unpackServerData(unpackedCt.start_confirmed_place),
+          end_confirmed_place: unpackServerData(unpackedCt.end_confirmed_place),
+          locations: unpackedCt.locations?.map(unpackServerData),
+          sections: unpackedCt.sections?.map(unpackServerData),
+        };
+      });
+    })
+    .catch((err) => {
+      displayError(err, 'while reading confirmed trips');
+      return [];
+    });
+};
+const dateTime2localdate = function (currtime: DateTime, tz: string) {
+  return {
+    timezone: tz,
+    year: currtime.get('year'),
+    //the months of the draft trips match the one format needed for
+    //moment function however now that is modified we need to also
+    //modify the months value here
+    month: currtime.get('month') + 1,
+    day: currtime.get('day'),
+    weekday: currtime.get('weekday'),
+    hour: currtime.get('hour'),
+    minute: currtime.get('minute'),
+    second: currtime.get('second'),
```

Comment:
I think you can access all of these directly (like `currtime.day`, `currtime.weekday`)